Fix clang-cl warnings of WebCore for Windows port

fujii opened the pull request:
#### afbcfab2cefd87cd5534a9a3086efc21e97cd727
<pre>
Fix clang-cl warnings of WebCore for Windows port
<a href="https://bugs.webkit.org/show_bug.cgi?id=259100">https://bugs.webkit.org/show_bug.cgi?id=259100</a>

Reviewed by Ross Kirsling.

clang-cl reported the following warnings:

&gt; platform\graphics\win\ComplexTextControllerUniscribe.cpp(141,9): warning: unused variable &apos;i&apos; [-Wunused-variable]
&gt; platform\graphics\egl\GLContext.cpp(228,29): warning: enumeration value &apos;Windows&apos; not handled in switch [-Wswitch]
&gt; platform\graphics\egl\GLContext.cpp(359,33): warning: enumeration value &apos;Windows&apos; not handled in switch [-Wswitch]
&gt; platform\graphics\egl\GLContext.cpp(406,33): warning: enumeration value &apos;Windows&apos; not handled in switch [-Wswitch]
&gt; platform\win\KeyEventWin.cpp(238,7): warning: field &apos;m_windowsVirtualKeyCode&apos; will be initialized after field &apos;m_autoRepeat&apos; [-Wreorder-ctor]
&gt; platform\win\PlatformMouseEventWin.cpp(86,7): warning: field &apos;m_buttons&apos; will be initialized after field &apos;m_clickCount&apos; [-Wreorder-ctor]
&gt; platform\win\WheelEventWin.cpp(78,7): warning: field &apos;m_globalPosition&apos; will be initialized after field &apos;m_directionInvertedFromDevice&apos; [-Wreorder-ctor]
&gt; platform\graphics\win\MediaPlayerPrivateMediaFoundation.cpp(84,13): warning: delete called on non-final &apos;WebCore::MediaPlayerPrivateMediaFoundation::AsyncCallback&apos; that has virtual functions but non-virtual destructor [-Wdelete-non-abstract-non-virtual-dtor]
&gt; platform\graphics\win\MediaPlayerPrivateMediaFoundation.cpp(2834,26): warning: unused variable &apos;image&apos; [-Wunused-variable]
&gt; Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp(154,31): warning: lambda capture &apos;this&apos; is not used [-Wunused-lambda-capture]
&gt; dom/FullscreenManager.cpp(299,65): warning: lambda capture &apos;this&apos; is not used [-Wunused-lambda-capture]
&gt; rendering\RenderTable.h(66,23): warning: inline function &apos;WebCore::RenderTable::borderTop&apos; is not defined [-Wundefined-inline]
&gt; rendering\RenderTable.h(67,23): warning: inline function &apos;WebCore::RenderTable::borderBottom&apos; is not defined [-Wundefined-inline]

* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::collectDataForWriting):
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::cancelFullscreen):
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
(WebCore::stringIndicesFromClusters):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
(WebCore::MediaPlayerPrivateMediaFoundation::AsyncCallback::~AsyncCallback):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::paintCurrentFrame):
* Source/WebCore/platform/win/KeyEventWin.cpp:
(WebCore::PlatformKeyboardEvent::PlatformKeyboardEvent):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/platform/win/WheelEventWin.cpp:
(WebCore::PlatformWheelEvent::PlatformWheelEvent):
* Source/WebCore/rendering/mathml/RenderMathMLBlockInlines.h:

Canonical link: <a href="https://commits.webkit.org/266001@main">https://commits.webkit.org/266001@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ba133432e4b45afe891ab8920204efa66fe9f07

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12833 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13147 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14245 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11999 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12563 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15314 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12852 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14688 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12667 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13446 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10574 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14670 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10723 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11317 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18411 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11810 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11481 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14673 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11966 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9891 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11205 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15536 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1410 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11821 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
```diff
@@ -156,7 +156,7 @@ void ClipboardItemBindingsDataSource::collectDataForWriting(Clipboard& destinati
 
             Ref itemTypeLoader { *weakItemTypeLoader };
-#if !COMPILER(MSVC)
+#if !COMPILER(MSVC) || COMPILER(CLANG)
```
It's kind of difficult to remember at a glance that COMPILER(MSVC) and COMPILER(CLANG) can both be true. If we don't want to add a new COMPILER, I think `!(COMPILER(MSVC) && !COMPILER(CLANG))` would at least make it clearer that it's not a typo.

fujii replied:
Fixed.
I might have taken a wrong decision. https://bugs.webkit.org/show_bug.cgi?id=171618#c10
Anyway, I hope we will remove MSVC support soon.

rkirsling commented on Source/WebCore/dom/FullscreenManager.cpp:
```diff
@@ -308,8 +308,13 @@ void FullscreenManager::cancelFullscreen()
             topDocument->page()->chrome().client().exitFullScreenForElement(fullscreenElement);
-        else
+        else {
+#if !RELEASE_LOG_DISABLED
```
This isn't the only `INFO_LOG` in this lambda, so it seems like putting
```
#if RELEASE_LOG_DISABLED
             UNUSED_PARAM(this);
#endif
```
at the beginning of the lambda would be more appropriate.

fujii replied:
Good point. Fixed. Thank you very much.

fujii committed:
Fix clang-cl warnings of WebCore for Windows port
https://bugs.webkit.org/show_bug.cgi?id=259100

Reviewed by Ross Kirsling.

clang-cl reported the following warnings:

> platform\graphics\win\ComplexTextControllerUniscribe.cpp(141,9): warning: unused variable 'i' [-Wunused-variable]
> platform\graphics\egl\GLContext.cpp(228,29): warning: enumeration value 'Windows' not handled in switch [-Wswitch]
> platform\graphics\egl\GLContext.cpp(359,33): warning: enumeration value 'Windows' not handled in switch [-Wswitch]
> platform\graphics\egl\GLContext.cpp(406,33): warning: enumeration value 'Windows' not handled in switch [-Wswitch]
> platform\win\KeyEventWin.cpp(238,7): warning: field 'm_windowsVirtualKeyCode' will be initialized after field 'm_autoRepeat' [-Wreorder-ctor]
> platform\win\PlatformMouseEventWin.cpp(86,7): warning: field 'm_buttons' will be initialized after field 'm_clickCount' [-Wreorder-ctor]
> platform\win\WheelEventWin.cpp(78,7): warning: field 'm_globalPosition' will be initialized after field 'm_directionInvertedFromDevice' [-Wreorder-ctor]
> platform\graphics\win\MediaPlayerPrivateMediaFoundation.cpp(84,13): warning: delete called on non-final 'WebCore::MediaPlayerPrivateMediaFoundation::AsyncCallback' that has virtual functions but non-virtual destructor [-Wdelete-non-abstract-non-virtual-dtor]
> platform\graphics\win\MediaPlayerPrivateMediaFoundation.cpp(2834,26): warning: unused variable 'image' [-Wunused-variable]
> Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp(154,31): warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
> dom/FullscreenManager.cpp(299,65): warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
> rendering\RenderTable.h(66,23): warning: inline function 'WebCore::RenderTable::borderTop' is not defined [-Wundefined-inline]
> rendering\RenderTable.h(67,23): warning: inline function 'WebCore::RenderTable::borderBottom' is not defined [-Wundefined-inline]

* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::collectDataForWriting):
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::cancelFullscreen):
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
(WebCore::stringIndicesFromClusters):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
(WebCore::MediaPlayerPrivateMediaFoundation::AsyncCallback::~AsyncCallback):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::paintCurrentFrame):
* Source/WebCore/platform/win/KeyEventWin.cpp:
(WebCore::PlatformKeyboardEvent::PlatformKeyboardEvent):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/platform/win/WheelEventWin.cpp:
(WebCore::PlatformWheelEvent::PlatformWheelEvent):
* Source/WebCore/rendering/mathml/RenderMathMLBlockInlines.h:

Canonical link: https://commits.webkit.org/266001@main
